src/components/ActivityFeed/ActivityFeed.js

```diff
@@ -87,46 +131,17 @@ Review.propTypes = {
   rating: number.isRequired,
 };
 
-// Check if a transition is the kind that
-// should be rendered in he ActivityFeed
-const shouldRenderTransition = transition => {
-  return [
-    TRANSITION_ACCEPT,
-    TRANSITION_CANCEL,
-    TRANSITION_COMPLETE,
-    TRANSITION_DECLINE,
-    TRANSITION_EXPIRE,
-    TRANSITION_REQUEST,
-    TRANSITION_REQUEST_AFTER_ENQUIRY,
-    TRANSITION_REVIEW_1_BY_CUSTOMER,
-    TRANSITION_REVIEW_1_BY_PROVIDER,
-    TRANSITION_REVIEW_2_BY_CUSTOMER,
-    TRANSITION_REVIEW_2_BY_PROVIDER,
-  ].includes(transition);
-};
-
-// Check if a user giving a review is related to
-// given tx transition.
-const isReviewTransition = transition => {
-  return [
-    TRANSITION_REVIEW_1_BY_CUSTOMER,
-    TRANSITION_REVIEW_1_BY_PROVIDER,
-    TRANSITION_REVIEW_2_BY_CUSTOMER,
-    TRANSITION_REVIEW_2_BY_PROVIDER,
-  ].includes(transition);
-};
-
-const hasUserLeftAReviewFirst = (userRole, lastTransition) => {
+const hasUserLeftAReviewFirst = (userRole, transaction, lastTransition) => {
```

Comment:
I think this should be separate utility function in util/transaction.js

Reply:
I think this would be more readable with variables:

```js
const hasUserLeftAReviewFirst = (userRole, transaction) => {
  const isProvider = txRoleIsProvider(userRole);
  return isProvider
    ? txHasFirstReviewBy(transaction, isProvider));
    : txHasFirstReviewBy(transaction, !isProvider));
};
```
then usage should be something like:

```js
    case TRANSITION_COMPLETE: {
      // Show the leave a review link if the state is delivered or
      // if current user is not the first to leave a review
      const reviewPeriodJustStarted = txIsDelivered(transaction);
      const reviewPeriodIsOver = txIsReviewed(transaction);
      const userHasLeftAReview = hasUserLeftAReviewFirst(ownRole, transaction);
      const reviewLink =
        reviewPeriodJustStarted || !(reviewPeriodIsOver || userHasLeftAReview) ? (
          <InlineTextButton onClick={onOpenReviewModal}>
            <FormattedMessage id="ActivityFeed.leaveAReview" values={{ displayName }} />
          </InlineTextButton>
        ) : null;

      return <FormattedMessage id="ActivityFeed.transitionComplete" values={{ reviewLink }} />;
    }
```
```js
        // show the leave a review link if current user is not the first
        // one to leave a review
        const reviewPeriodIsOver = txIsReviewed(transaction);
        const userHasLeftAReview = hasUserLeftAReviewFirst(ownRole, transaction);
        const reviewLink = !(reviewPeriodIsOver || userHasLeftAReview) ? (
```